tasks/update-ocp-tag/update-ocp-tag.yaml

```
@@ -73,6 +75,9 @@ spec:
         # if {{OCP_VERSION}} is not set, the original targetIndex will be kept but its ocp version should
         # match base image version.
         for index in "${updatedFromIndex}" "${updatedTargetIndex}" "${updatedBinaryImage}"; do
+          if [[ "$stagedIndex" == "true" and "$index" == "" ]]; then
+            continue
+          fi
           validateOCPVersion "${index}" "$(params.ocpVersion)"
         done
```

I wonder if it would make more sense to just call them one by one, since it's only 3 and the validation can only be skipped for targetIndex. And we don't even need stagedIndex.
```suggestion
        validateOCPVersion "${updatedFromIndex}" "$(params.ocpVersion)"
        validateOCPVersion "${updatedBinaryImage}" "$(params.ocpVersion)"
        if [ -n "${updatedTargetIndex}" ]; then
            validateOCPVersion "${updatedTargetIndex}" "$(params.ocpVersion)"
        fi
```

---

true. going for it.